### ISSUE TITLE
"RegisteredDomains Response is Null in the .NetClient" Issue And example Code for .Net Forms

### ISSUE THREAD

**Issue by Maximilian-Rief** [jliliamen/WordPressLicenseManagerNETClient#15]
Hi, 
first of all thanks for your work, it was really what I was looking for.
I have impremented an example code for .Net Forms. Everything seemed to work. I have included the Hardware Id in RegisteredDomains but after deleting the RegisteredDomain entry and adding it again something happened. I think the database have some problem, I am not sure of what it is, but now RegisteredDomains responds null. I might have done some stupid mistake in my code I don't know.
Do you know what the problem could be? (I have used this wordpress plugin first https://github.com/michelve/software-license-manager. This might be the issue... I have tried uninstalling it and installing your suggested one but it still doesn't work.) 
In the meantime I am going to share the example code for .Net. Btw it works perfectly if not for the Hardware Id Part. All that Messageboxes are just for debugging purposes sorry for that :)

```
WordPressLicenseManagerNETClient.Models.License license = new WordPressLicenseManagerNETClient.Models.License();
        ILicenseManager licenseManager = default(ILicenseManager);
        Configuration configuration = default(Configuration);    
public enum License_Request_result
        {
            Success = 0,
            Invalid_key,
            Expired,
            Blocked,
            Pending,
            No_Internet,
            Max_Hwid_Reached,
            Unknown,
            Input_hwid_null,
            Input_license_null
        }
        private License_Request_result check_license(String license_key, String hwid)
        {
            if (hwid == "")
                return License_Request_result.Input_hwid_null;
            if (license_key == "")
                return License_Request_result.Input_license_null;
            configuration = new Configuration();
            configuration.PostURL = "https://mysite.com";
            configuration.ActivationKey = "activation123";
            licenseManager = LicenseManagerFactory.New(configuration);
            license.Key = license_key;
            var licenseResponse = licenseManager.PerformAction(WordPressLicenseManagerNETClient.Consts.Action.Check, license);
            if (licenseResponse == null) //No Internet
            {
                //MessageBox.Show("No internet");
                return License_Request_result.No_Internet;
            }
            else
            {
                if (licenseResponse.Success == false)
                    //throw new Exception(licenseResponse.Message);
                    return License_Request_result.Invalid_key;
                else
                {
                    if(licenseResponse.RegisteredDomains == null)
                    {
                        MessageBox.Show("NUll registred");
                    }
                   // Console.WriteLine("Response: " + licenseResponse.RegisteredDomains.);
                    MessageBox.Show("Status is true: " + licenseResponse.Status);
                    switch (licenseResponse.Status)
                    {
                        case "active":
                            {
                                Console.WriteLine("Server Response: " + licenseResponse.Message);
                                MessageBox.Show("Registered Hwids: " + licenseResponse.RegisteredDomains.Count);
                                foreach (var domain in licenseResponse.RegisteredDomains)
                                {
                                    Console.WriteLine($"Device Name: {domain.Value}");
                                }
                                var found_my_current_hwid = licenseResponse.RegisteredDomains.Find(domain => domain.Value == hwid); //checks if my hardware id is in the list
                                if(found_my_current_hwid == null)
                                {
                                    MessageBox.Show("adding my hwid");
                                    license.RegisterDomain(hwid);
                                    licenseResponse = null;
                                     licenseResponse = licenseManager.PerformAction(WordPressLicenseManagerNETClient.Consts.Action.Activate, license);
                                    if (licenseResponse.Success == false)
                                    {
                                        Console.WriteLine(licenseResponse.Message);
                                        if(licenseResponse.Message == "Reached maximum allowable domains")
                                        {
                                            return License_Request_result.Max_Hwid_Reached;
                                        }
                                        MessageBox.Show("Unhandled Response when adding Hwid: "+ licenseResponse.Message + "| Report this message to our Support please.");
                                        return License_Request_result.Unknown;
                                    }
                                    else
                                    {
                                        Console.WriteLine("successfully added my hwid" + hwid +" "+ licenseResponse.Message);
                                    }
                                }
                                else
                                {
                                    Console.WriteLine("found Hwid: " + found_my_current_hwid.Value);
                                }
                                return License_Request_result.Success;
                            }
                        case "pending":
                            return License_Request_result.Pending;
                        case "expired":
                            return License_Request_result.Expired;
                        case "blocked":
                            return License_Request_result.Blocked;
                        default:
                            return License_Request_result.Unknown;
                    }
                    //  Assert.IsTrue(licenseResponse.Success, $"Key {licenseResponse.Key}: Domains: {string.Join(", ", licenseResponse.RegisteredDomains.Select(x => x.Value).ToArray())}");
                }
            }
        }
        private void button1_Click(object sender, EventArgs e)
        {
            String hwid_string = Environment.MachineName + Environment.UserName;
            MessageBox.Show("hwid: " + hwid_string);
            MessageBox.Show(check_license("123", hwid_string);).ToString());

        }
```

**Comment by AmenJlili** [comment]
If you delete RegisteredDomain from WP and there is no other one, RegisteredDomains would return null in the license response. I'm not too sure if I understand you correctly...

**Comment by Maximilian-Rief** [comment]
I forot to mention sorry. of course there is the domain entry. I see it on wordpress, but not on the .net client

**Comment by AmenJlili** [comment]
Hmm.  I think I might have seen this before but never bothered to fix it. Try calling the check enum after you activate. It will definitely list the registered domains there and you can leverage that in the rest of your code.

**Comment by Maximilian-Rief** [comment]
Thanks for the help. But I think I will just leave it like that. I just realized I can just call Activate, and if the hwid limit is exeeded it will detect that. not optimal but it works

**Comment by AmenJlili** [comment]
@Maximilian-Rief  Impressive work by a 20-year-old! Bonne continuation!

**Comment by Maximilian-Rief** [comment]
> @Maximilian-Rief Impressive work by a 20-year-old! Bonne continuation!

Thank you. Have a good day :)

**Comment by Maximilian-Rief** [comment]
I actually figured out what it was for me. The **wordpress cache** was enabled, I think it was returning the cached value. I cleaned the cache right after adding the hwid and it refreshes instantly! Same with the licenses. 
With cache on it still works, **but it needs more time to refresh**, that's it.
In the future I will try to create an exception for the plugin so this doesn't happen. Best scenario would be having the cache on and excluding the license plugin...
